mapeditor/inspector/townbuildingswidget.cpp

```diff
@@ -353,11 +353,41 @@ void TownBuildingsDelegate::setModelData(QWidget *editor, QAbstractItemModel *mo
 		town.removeAllBuildings();
 		for(const auto & building : ed->getBuiltBuildings())
 			town.addBuilding(building);
+		updateModelData(model, index);
 	}
 	else
 	{
 		QStyledItemDelegate::setModelData(editor, model, index);
 	}
 }
 
-
+void TownBuildingsDelegate::updateModelData(QAbstractItemModel * model, const QModelIndex & index) const
+{
+	QStringList textList;
+	textList += QObject::tr("Built buildings:");
+	auto * ctown = town.getTown();
+	if(!ctown)
+		ctown = VLC->townh->randomTown;
```

Comment:
is it an often used pattern? If yes, better extract to a function

Reply:
Should be no longer necessary. Simple getTown should suffice:
```suggestion
	auto * ctown = town.getTown();
```

`getTown()` method now returns random town for well, random towns.